천수빈_PR

b1ctory opened the pull request:
커밋 로그를 기능별로 작성하고 수정하면서 refactor 로그까지 남겨놨었는데 pr 버튼 활성화가 되지 않아서

다시 fork 후 로그를 작성하여 기능별 세부 로그가 남아있지 않습니다 😢 

마구마구 지적해주세요 🔥

b1ctory committed:
initial commit

b1ctory committed:
add Department enum class

b1ctory committed:
add Employee class and inheritance classes

Employee 클래스와 상속받는 클래스들을 선언

b1ctory committed:
add Employee Manager

zeratulspc commented on InheritanceReal/src/EmployeeManager.kt:
```diff
@@ -0,0 +1,229 @@
+
+        println("\nPlease check the input information. >> ")
+        println("Name : ${employee.name} \nID : ${employee.id} \nDepartment : ${employee.department.departmentName} \n" +
+                "Monthly Salary : ${employee.calculateSalary()}만원")
```
NumberFormat 클래스의 getCurrencyInstance()를 활용해보세요!

zeratulspc commented on InheritanceReal/src/EmployeeManager.kt:
```diff
@@ -0,0 +1,229 @@
+    }
+
+    // Remove Employee
+    private fun removeEmployeeByName() {
+
+        if (employees.isEmpty()) {
+            println("There is nothing to Remove !!! ")
+        } else {
+            print("Enter the name of the employee you want to remove. >> ")
+
+            val inputName = readLine()
+            if (!employees.contains(inputName)) {
+                println("Wrong Name Entered !!!")
+            } else {
+                for (employee in employees) {
+                    if (employee.name == inputName) {
+                        employees.remove(employee)
+                        println("====== Successfully Removed =======\n")
+                        break
+                    }
+                }
+            }
+
+        }
+
+    }
```
중복된 이름이 있는 경우를 고려해서
uuid로 검색 후 제거하는것도 나쁘지 않아 보여요 😃

b1ctory replied:
uuid는 랜덤값으로 생성되는 것이라 제외했는데 중복 이름은 고려를 못헀네요! 
두가지를 함께 넣어서 제거하는 방식으로 수정해볼게요 ~ 😄

zeratulspc commented on InheritanceReal/src/EmployeeManager.kt:
```diff
@@ -0,0 +1,229 @@
+    }
+
+    // Average of Employees Salary
+    private fun averageOfEmployeesSalary() {
```
함수 이름 좋아요!!!!
이것이 짬에서 나오는 바이브인가..? 😮

zeratulspc reviewed (commented):
코드리뷰 남깁니다!

b1ctory committed:
fix currency format

만원 단위로 표시하던 방식을 getCurrencyInstance() 메소드 활용하는 방식으로 수정

b1ctory committed:
fix remove method

기존 employee 제거 방식은 이름만을 이용한 방식이었지만, 피드백 반영하여 (중복 이름 고려) name과 id를 두가지 모두 받아서 제거하는 방식으로 수정

zeratulspc commented on InheritanceReal/src/Employee.kt:
```diff
@@ -0,0 +1,33 @@
+    companion object {
+        const val BASIC_YEARLY_SALARY = 12000
+        const val BASIC_DAILY_SALARY = 2.5
+    }
```
주석을 넣어볼까요?? 😄 👍

b1ctory committed:
refoator Employee Manager

b1ctory committed:
add remark
